airflow/config_templates/config.yml

```
@@ -366,6 +366,17 @@
       type: string
       example: "path.to.CustomXCom"
       default: "airflow.models.xcom.BaseXCom"
+    - name: additional_execute_contextmanager
+      description: |
+        Custom user function that returns a context manager. Syntax is "package.method".
+        Context is entered when operator starts executing task. __enter__() will be called
+        before the operator's execute method, and __exit__() shortly after.
+        Function's signature should accept two positional parameters - task instance
+        and execution context
```

```suggestion
        Context is entered when operator starts executing task. ``__enter__()`` will be called
        before the operator's execute method, and ``__exit__()`` shortly after.
        Function's signature should accept two positional parameters - task instance
        and execution context
```

I think you can add what it is equivalent to, to make it clear:

```python
@my_context_manager
def execute(context):
    ...
```

---

if not here, we should add it to `docs/howto/use-additional-execute-contextmanager.rst`

---

I'm not sure that clears things up -- personally it took me a few minutes to understand myself how this is equivalent.
I think the simplistic explanation is more understandable than the code example.

---

Sure, it might just be me